src/core/system_parameters/beidou_cnav1_navigation_message.cc

```diff
@@ -0,0 +1,605 @@
+
+    // Only include data without 3 lsb from crc
+    Crc_Beidou_Cnav1.process_bytes(bytes.data(), BEIDOU_CNAV1_DATA_BYTES - 3);
+    crc_computed = Crc_Beidou_Cnav1.checksum();
+
+    if (crc_decoded == crc_computed)
+        {
+            return true;
+        }
+    else
+        {
+            return false;
+        }
+}
+
+
+int32_t Beidou_Cnav1_Navigation_Message::frame_decoder(std::string const &frame_string)
+{
+	// Gets the message data which contains Subframe 1,Subframe 2,Subframe 3
+    	std::bitset<BEIDOU_CNAV1_DATA_BITS> frame_bits(frame_string);
+    
+        std::string subframe1_string;   					//string for storing subframe 1
+	std::string subframe2_string;						//string for storing subframe 2
+	std::string subframe3_string;						//string for storing subframe 3
+	
+	
+	subframe1_string=frame_string.substr(0, BEIDOU_CNAV1_SUBFRAME_1_BITS);
+	subframe2_string=frame_string.substr(14, BEIDOU_CNAV1_SUBFRAME_2_BITS);
+	subframe3_string=frame_string.substr(614, BEIDOU_CNAV1_SUBFRAME_3_BITS);
+	
+	std::bitset<BEIDOU_CNAV1_SUBFRAME_1_BITS> subframe1(subframe1_string);  //Gets the message data which contains Subframe 1
+	std::bitset<BEIDOU_CNAV1_SUBFRAME_2_BITS> subframe2(subframe2_string);  //Gets the message data which contains Subframe 2
+	std::bitset<BEIDOU_CNAV1_SUBFRAME_3_BITS> subframe3(subframe3_string);  //Gets the message data which contains Subframe 3
+    
+    
+    	
+    	/* TO DO: Checking CRC for both subframes
+    	//! CNAV1 navigation message have CRC in two subframes i.e in Subframe 2 and Subframe 3
+    	
+    	// Gets the crc data for comparison, last 24 bits from 600 bit of Subframe 2
+    	std::bitset<BEIDOU_CNAV1_CRC_BITS> checksum_sf_2(subframe2.substr(576, 24));
+    	// Gets the crc data for comparison, last 24 bits from 264 bit of Subframe 3
+    	std::bitset<BEIDOU_CNAV1_CRC_BITS> checksum_sf_3(subframe3.substr(240, 24));
+
+    	// Perform data verification and exit code if error in bit sequence
+    	flag_crc_test_sf_2 = crc_test(subframe2, checksum_sf_2.to_ulong());
+    	flag_crc_test_sf_3 = crc_test(subframe3, checksum_sf_3.to_ulong());
+
+    	if (flag_crc_test_sf_2 == false && flag_crc_test_sf_3 == false)
+        	return 0;
+        */
+        
+        // Gets the crc data for comparison, last 24 bits from 288 bit data frame
+    	std::bitset<BEIDOU_CNAV1_CRC_BITS> checksum(frame_string.substr(854, 24));
+
+    	// Perform data verification and exit code if error in bit sequence
+    	flag_crc_test = crc_test(frame_bits, checksum.to_ulong());
+
+    	if (flag_crc_test == false)
+        	return 0;
+    	
+    	//===============================================Decode string message of Subframe 1==============================================
+    	
+    	cnav1_ephemeris.i_satellite_PRN = static_cast<uint32_t>(read_navigation_unsigned(subframe1, PRN));
+    	cnav1_ephemeris.SOH = static_cast<uint32_t>(read_navigation_unsigned(subframe1, SOH)) * 18;   ///Scalling factor= 18, Unit= [s]
```

Comment:
Make the 18 a constant rather than a hardcoded number

Reply:
Replaced hardcoded number by constant.